cmd/tools/rest/HrefBuilder.go

```diff
@@ -0,0 +1,79 @@
+package rest
+
+import "strings"
+
+type HrefBuilder struct {
+	apiPath       string
+	fields        string
+	filter        []string
+	queryFields   string
+	queryValue    string
+	maxRecords    string
+	returnTimeout string
+}
+
+func NewHrefBuilder() *HrefBuilder {
+	return &HrefBuilder{}
+}
+
+func (b *HrefBuilder) APIPath(apiPath string) *HrefBuilder {
+	b.apiPath = apiPath
+	return b
+}
+
+func (b *HrefBuilder) Fields(fields string) *HrefBuilder {
+	b.fields = fields
+	return b
+}
+
+func (b *HrefBuilder) Filter(filter []string) *HrefBuilder {
+	b.filter = filter
+	return b
+}
+
+func (b *HrefBuilder) QueryFields(queryFields string) *HrefBuilder {
+	b.queryFields = queryFields
+	return b
+}
+
+func (b *HrefBuilder) QueryValue(queryValue string) *HrefBuilder {
+	b.queryValue = queryValue
+	return b
+}
+
+func (b *HrefBuilder) MaxRecords(maxRecords string) *HrefBuilder {
+	b.maxRecords = maxRecords
+	return b
+}
+
+func (b *HrefBuilder) ReturnTimeout(returnTimeout string) *HrefBuilder {
```

Comment:
Should this be a go duration? I can't find any references to return_timeout to check